cheatsheet.js

```diff
@@ -0,0 +1,300 @@
+// The code inside the curly brackets will not execute until you "call" this function!
+// You can declare a function like this:
+function myFunctionName() {
+  console.log('Called this function!');
+}
+
+// The above *technically* is shorthad for this (you typically won't do this):
+var myFunctionName = function() {
+  console.log('Called this function!');
+}
+
+// Functions are "called" using `()`. You call functions like this:
+myFunctionName(); // Logs 'Called this function'
+
+// You can try to call things that aren't functions, but it won't go well...
+var notAFunction = false;
+notAFunction(); // Throws error: "notAFunction" is not a function.
+
+// Functions may take any number of arguments. You can pass values to the
+// function inside the `()` used to call it! Those values will be available
+// for use inside the function's "scope".
+function iTakeArguments(arg1, arg2, arg3){
+  console.log(arg1, arg2, arg3);
+}
+
+var val = 'value';
+iTakeArguments(val, true, 10); // Logs: 'value' true 10
+iTakeArguments(val, false, 20); // Logs: 'value' false 20
+
+// Functions may "return" a value when called. This value can be used like any
+// other variable.
+function timesTwo(num) {
+  return num * 2;
+}
+var a = timesTwo(2);
+console.log(a) // Logs: 4
+console.log(timesTwo(4)) // Logs: 8
+
+// Becuase functions are just fancy values that can be "called", it stands to
+// reason that functions can be passed around like other variables:
+function callThis(func){
+  return func();
+}
+function one(){
+  return 1;
+}
+function two() {
+  return 1;
+}
+console.log(callThis(one)) // Logs: 1
+console.log(callThis(two)) // Logs: 2
+
+// Also because functions are just fancy values, it turns our that functions
+// can be assigned as values on an object! When a function is stored in an
+// object, it is said to be a "method" of that object. For example, this
+// object has a "method" called `notify`:
+var obj = {
+  notify(){
+    console.log('Notification!');
+  }
+  // The above is fancy shorthand for this:
+  // notify: function(){ console.log('Notification!'); }
+};
+
+// You may call this object's method by accessing it using either dot or bracket
+// notation, and calling it using `()` like we do above.
+obj.notify(); // Logs: Notification!
+
+// I mentioned the word "scope" above. Scope is tricky! Scope is all about
+// what variables are available to a line of code. Functions always define a
+// new scope. Any variables defined in this scope are available to itself, and
+// any other scopes delcared within it. If a variable is with the same name
+// is re-declared in a different scope, it uses the closest declared variable.
+// For example:
+var a = 1;
+
+function one() {
+  var b = 2;
+
+  function two() {
+    var a = 4;
+    var c = 3;
+    console.log(a, b, c) // Logs: 4, 2, 3
+  }
+
+  two();
+  console.log(a, b) // Logs: 1, 2, undefined
+}
+
+console.log(a, b, c); // Logs 1, undefined, undefined
+one();
```

Comment:
TODO: Add control blocks here! (Like `if`, `else if`, `else`, `while`, `for`, etc...)